anndata/_core/sparse_dataset.py

```diff
@@ -343,7 +343,29 @@ def __repr__(self) -> str:
     def __getitem__(self, index: Index | tuple[()]) -> float | ss.spmatrix:
         row, col = self._normalize_index(index)
         mtx = self._to_backed()
-        sub = mtx[row, col]
+
+        # see https://github.com/scverse/anndata/issues/1224 for special handling of boolean masks
+        # scipy internally converts boolean masks to integer indices so this needs to occur before access to `mtx`.
+        def make_slices(mask):
+            return np.ma.extras._ezclump(mask)
+
+        def mean_slice_length(slices):
+            return floor((slices[-1].stop - slices[0].start) / len(slices))
```

Comment:
Can we assume the `slices` are sorted? I think so?

Reply:
Because we made them, yes.